### PR TITLE
Source Hubspot: add integration tests

### PR DESCRIPTION
**Pull request by davydov-d**
What
The issue: https://github.com/airbytehq/airbyte/issues/35188

How
Cover streams defined as empty in acceptance-test-config.yaml with integration tests

### CODE REVIEW

**Review by bazarnov — APPROVED** [review]

**Comment by davydov-d** [comment]
Hey @maxi297 can you please review this PR? Thanks!

File: airbyte-integrations/connectors/source-hubspot/unit_tests/integrations/test_web_analytics_streams.py [hunk 1]
```diff
@@ -0,0 +1,567 @@
...
+        )
+        output = self.read_from_stream(self.private_token_config(self.ACCESS_TOKEN), stream_name, SyncMode.full_refresh)
+        record = output.records[0].record.data
+        assert "properties" not in record
+        prop_fields = len([f for f in record if f.startswith("properties_")])
+        assert prop_fields > 0
+
+    @pytest.mark.parametrize(("stream_name", "parent_stream_name", "object_type", "parent_stream_associations"), CRM_STREAMS)
+    @HttpMocker()
+    def test_given_one_page_when_read_then_get_no_records_filtered(
```
Comment (maxi297) [inline airbyte-integrations/connectors/source-hubspot/unit_tests/integrations/test_web_analytics_streams.py:323]: Why would they be filtered?

**Review by maxi297 — COMMENTED** [review]
I like the coverage the tests provide. My concerns are only about expressiveness of the tests and how well they document the behavior. As such, I don't think I have hard blockers but I would like to get a feeling or how we can improve them (even if we don't take action). Feel free to challenge them and I'll approve after!

File: airbyte-integrations/connectors/source-hubspot/unit_tests/integrations/test_web_analytics_streams.py (hunk 1 above)
Comment (davydov-d) [reply airbyte-integrations/connectors/source-hubspot/unit_tests/integrations/test_web_analytics_streams.py:367]: we have a [filter](https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/connectors/source-hubspot/source_hubspot/streams.py#L330) in one of the base classes that can be turned on or off per class, so this test is checking that this specific stream does not filter out the records

File: airbyte-integrations/connectors/source-hubspot/unit_tests/integrations/test_web_analytics_streams.py [hunk 2]
```diff
@@ -0,0 +1,567 @@
...
+                start, end = dt_range
+                web_analytics_response = self.web_analytics_response(stream_name)
+                self.mock_response(
+                    http_mocker,
+                    self.web_analytics_request(stream_name, self.ACCESS_TOKEN, _id, object_type, start, end),
+                    web_analytics_response
+                )
+        config_start_dt = date_ranges[0][0]
+        output = self.read_from_stream(self.private_token_config(self.ACCESS_TOKEN, config_start_dt), stream_name, SyncMode.full_refresh)
+        assert len(output.records) == 4
```
[the remark this answers is not in the document]
Comment (davydov-d) [reply airbyte-integrations/connectors/source-hubspot/unit_tests/integrations/test_web_analytics_streams.py]: 2 parent records * 2 datetime slices -- I added a comment to explain that

File: airbyte-integrations/connectors/source-hubspot/unit_tests/integrations/test_web_analytics_streams.py [hunk 3]
```diff
@@ -0,0 +1,567 @@
...
+            http_mocker,
+            self.web_analytics_request(stream_name, self.ACCESS_TOKEN, self.OBJECT_ID, object_type),
+            self.web_analytics_response(stream_name)
+        )
+        output = self.read_from_stream(self.private_token_config(self.ACCESS_TOKEN), stream_name, SyncMode.full_refresh)
+        assert len(output.records) == 1
+
+    @pytest.mark.parametrize(("stream_name", "parent_stream_name", "object_type", "parent_stream_associations"), CRM_STREAMS)
+    @HttpMocker()
+    def test_given_two_pages_when_read_then_return_records(
```
[the remark this answers is not in the document]
Comment (davydov-d) [reply airbyte-integrations/connectors/source-hubspot/unit_tests/integrations/test_web_analytics_streams.py:164]: yes, done

File: airbyte-integrations/connectors/source-hubspot/unit_tests/integrations/test_owners_archived.py [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+    stream is identical to the `Owners` stream (which is covered by acceptance tests), except for a single url param.
+    """
+    SCOPES = ["crm.objects.owners.read"]
+    CURSOR_FIELD = "updatedAt"
+    STREAM_NAME = "owners_archived"
+
+    def request(self):
+        return OwnersArchivedStreamRequestBuilder().build()
+
+    def response(self, with_pagination: bool = False):
```
[the remark this answers is not in the document]
Comment (davydov-d) [reply airbyte-integrations/connectors/source-hubspot/unit_tests/integrations/test_owners_archived.py:30]: now it is :)

File: airbyte-integrations/connectors/source-hubspot/unit_tests/integrations/test_engagements_calls.py [hunk 5]
```diff
@@ -0,0 +1,142 @@
...
+    @HttpMocker()
+    def test_given_one_page_when_read_then_get_transformed_records(self, http_mocker: HttpMocker):
+        self.mock_custom_objects(http_mocker)
+        self.mock_properties(http_mocker, self.OBJECT_TYPE, self.PROPERTIES)
+        self.mock_response(http_mocker, self.request(), self.response())
+        output = self.read_from_stream(self.private_token_config(self.ACCESS_TOKEN), self.STREAM_NAME, SyncMode.full_refresh)
+        record = output.records[0].record.data
+        assert "properties" in record  # legacy struct remains to not introduce breaking changes
+        prop_fields = len([f for f in record if f.startswith("properties_")])
+        assert prop_fields > 0
```
[the remark this answers is not in the document]
Comment (davydov-d) [reply airbyte-integrations/connectors/source-hubspot/unit_tests/integrations/test_engagements_calls.py:144]: yep, done

File: airbyte-integrations/connectors/source-hubspot/unit_tests/integrations/test_engagements_calls.py [hunk 6]
```diff
@@ -0,0 +1,142 @@
...
+            ]
+        )
+        with mock.patch("time.sleep"):
+            output = self.read_from_stream(self.private_token_config(self.ACCESS_TOKEN), self.STREAM_NAME, SyncMode.full_refresh)
+        assert len(output.records) == 1
+        assert len(output.trace_messages) > 0
+        assert len(output.errors) == 0
+
+    @HttpMocker()
+    def test_given_missing_scopes_error_when_read_then_hault(self, http_mocker: HttpMocker):
```
[the remark this answers is not in the document]
Comment (davydov-d) [reply airbyte-integrations/connectors/source-hubspot/unit_tests/integrations/test_engagements_calls.py]: We're validating that the connector stops the sync and does not continue making requests in an infinite loop or something. I've changed the name of the test.

File: airbyte-integrations/connectors/source-hubspot/unit_tests/integrations/test_engagements_calls.py [hunk 7]
```diff
@@ -0,0 +1,142 @@
...
+    @HttpMocker()
+    def test_given_one_page_when_read_stream_private_token_then_return_records(self, http_mocker: HttpMocker):
+        self.mock_custom_objects(http_mocker)
+        self.mock_properties(http_mocker, self.OBJECT_TYPE, self.PROPERTIES)
+        self.mock_response(http_mocker, self.request(), self.response())
+        output = self.read_from_stream(self.private_token_config(self.ACCESS_TOKEN), self.STREAM_NAME, SyncMode.full_refresh)
+        assert len(output.records) == 1
+
+    @HttpMocker()
+    def test_given_two_pages_when_read_then_return_records(self, http_mocker: HttpMocker):
```
[the remark this answers is not in the document]
Comment (davydov-d) [reply airbyte-integrations/connectors/source-hubspot/unit_tests/integrations/test_engagements_calls.py:82]: Done

File: airbyte-integrations/connectors/source-hubspot/unit_tests/integrations/test_engagements_calls.py [hunk 8]
```diff
@@ -0,0 +1,142 @@
...
+        ).with_field(
+            FieldPath("id"), self.OBJECT_ID
+        )
+        response = response_builder.with_record(record)
+        if with_pagination:
+            response = response.with_pagination()
+        return response.build()
+
+    @HttpMocker()
+    def test_given_one_page_when_read_stream_oauth_then_return_records(self, http_mocker: HttpMocker):
```
[the remark this answers is not in the document]
Comment (davydov-d) [reply airbyte-integrations/connectors/source-hubspot/unit_tests/integrations/test_engagements_calls.py]: We need to `mock_custom_objects` as there's a source level request being made before actually running the sync, so without mocking it, no test would pass.
Indeed, `self.ACCESS_TOKEN` is used to retrieve scopes.
I've split this test into two as you suggested, please see if you like it

**Comment by davydov-d** [comment]
@maxi297 thanks for the review! I've pushed some updates and replied to your comments, please take a look one more time

**Review by maxi297 — APPROVED** [review]
Cool! Thanks for addressing my comments. I think those tests are a great addition to the connector code!

**Pull request merged**